src/test/java/org/jenkinsci/plugins/workflow/cps/SnippetizerTest.java

```diff
@@ -125,34 +154,43 @@
         BuildTriggerStep step = new BuildTriggerStep("downstream");
         assertRoundTrip(step, "build 'downstream'");
         step.setParameters(Arrays.asList(new StringParameterValue("branch", "default"), new BooleanParameterValue("correct", true)));
+        // when this test fails because of new core, see below "buildTriggerStepWith_2_2"
         assertRoundTrip(step, "build job: 'downstream', parameters: [[$class: 'StringParameterValue', name: 'branch', value: 'default'], [$class: 'BooleanParameterValue', name: 'correct', value: true]]");
     }
 
+    @Test @Ignore("until this module can be compiled with 2.2 core")
+    public void buildTriggerStepWith_2_2() throws Exception {
+        BuildTriggerStep step = new BuildTriggerStep("downstream");
+        assertRoundTrip(step, "build 'downstream'");
+        step.setParameters(Arrays.asList(new StringParameterValue("branch", "default"), new BooleanParameterValue("correct", true)));
+        assertRoundTrip(step, "build job: 'downstream', parameters: [string(name: 'branch', value: 'default'), booleanParam(name: 'correct', value: true)]");
+    }
+
     @Issue("JENKINS-25779")
     @Test public void defaultValues() throws Exception {
         assertRoundTrip(new InputStep("Ready?"), "input 'Ready?'");
     }
 
     private static void assertRoundTrip(Step step, String expected) throws Exception {
         assertEquals(expected, Snippetizer.object2Groovy(step));
-        GroovyShell shell = new GroovyShell(r.jenkins.getPluginManager().uberClassLoader);
-        final StepDescriptor desc = step.getDescriptor();
-        shell.setVariable("steps", new GroovyObjectSupport() {
-            @Override public Object invokeMethod(String name, Object args) {
-                if (name.equals(desc.getFunctionName())) {
-                    try {
-                        return desc.newInstance(DSL.parseArgs(desc, args).namedArgs);
-                    } catch (RuntimeException x) {
-                        throw x;
-                    } catch (Exception x) {
-                        throw new RuntimeException(x);
-                    }
-                } else {
-                    return super.invokeMethod(name, args);
+        CompilerConfiguration cc = new CompilerConfiguration();
+        cc.setScriptBaseClass(DelegatingScript.class.getName());
+        GroovyShell shell = new GroovyShell(r.jenkins.getPluginManager().uberClassLoader,new Binding(),cc);
+
+        DelegatingScript s = (DelegatingScript) shell.parse(expected);
+        s.o = new DSL(new DummyOwner()) {
+            // for testing, instead of executing the step just return an instantiated Step
```

Comment:
(1) it doesn't work with nested use of symbols, and (2) duplicate code that interprets method name.